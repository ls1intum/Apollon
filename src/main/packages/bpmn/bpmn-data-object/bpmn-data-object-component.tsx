import React, { FunctionComponent } from 'react';
import { ThemedPolyline } from '../../../components/theme/themedComponents';
import { Multiline } from '../../../utils/svg/multiline';
import { BPMNDataObject } from './bpmn-data-object';

export const BPMNDataObjectComponent: FunctionComponent<Props> = ({ element, fillColor, strokeColor, textColor }) => (
  <g>
    <ThemedPolyline
      points={`0 0, 0 ${element.bounds.height}, ${element.bounds.width} ${element.bounds.height}, ${
        element.bounds.width
      } 15, ${element.bounds.width - 15} 0, ${element.bounds.width - 15} 15, ${element.bounds.width} 15, ${
        element.bounds.width - 15
      } 0, 0 0`}
      strokeColor={strokeColor || element.strokeColor}
      fillColor={fillColor || element.fillColor}
    />
    <Multiline
      x={element.bounds.width / 2}
      y={element.bounds.height + 20}
      width={element.bounds.width * 2}
      fill={textColor || element.textColor}
      lineHeight={16}
      capHeight={11}
      verticalAnchor="start"
    >
      {element.name}
    </Multiline>
  </g>
);

interface Props {
  element: BPMNDataObject;
  fillColor?: string;
  strokeColor?: string;
  textColor?: string;
}
