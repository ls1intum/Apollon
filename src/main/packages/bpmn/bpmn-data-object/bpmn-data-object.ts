import { BPMNElementType } from '..';
import { ILayer } from '../../../services/layouter/layer';
import { ILayoutable } from '../../../services/layouter/layoutable';
import { calculateNameBounds } from '../../../utils/name-bounds';
import { UMLElementType } from '../../uml-element-type';
import { UMLElementFeatures } from '../../../services/uml-element/uml-element-features';
import { UMLElement } from '../../../services/uml-element/uml-element';
import { UMLContainer } from '../../../services/uml-container/uml-container';

export class BPMNDataObject extends UMLContainer {
  static features: UMLElementFeatures = { ...UMLContainer.features };

  type: UMLElementType = BPMNElementType.BPMNDataObject;

  render(canvas: ILayer): ILayoutable[] {
    this.bounds = calculateNameBounds(this, canvas);
    return [this];
  }
}
