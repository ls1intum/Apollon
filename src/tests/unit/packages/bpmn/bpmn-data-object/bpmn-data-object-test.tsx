import { wrappedRender } from '../../../test-utils/render';
import * as React from 'react';
import { BPMNDataObject } from '../../../../../main/packages/bpmn/bpmn-data-object/bpmn-data-object';
import { BPMNDataObjectComponent } from '../../../../../main/packages/bpmn/bpmn-data-object/bpmn-data-object-component';

it('render the bpmn-transaction-component', () => {
  const dataObject: BPMNDataObject = new BPMNDataObject({ name: 'Data Object' });
  const { getByText, baseElement } = wrappedRender(
    <svg>
      <BPMNDataObjectComponent element={dataObject} />
    </svg>,
  );
  expect(getByText(dataObject.name)).toBeInTheDocument();
  expect(baseElement).toMatchSnapshot();
});
