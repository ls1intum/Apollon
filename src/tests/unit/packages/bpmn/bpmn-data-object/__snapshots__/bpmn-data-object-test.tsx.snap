// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the bpmn-transaction-component 1`] = `
<body>
  <div>
    <svg>
      <g>
        <polyline
          class="sc-aXZVg bUvEJV"
          fill="white"
          points="0 0, 0 100, 200 100, 200 15, 185 0, 185 15, 200 15, 185 0, 0 0"
          stroke="black"
        />
        <text
          font-weight="bold"
          height="100"
          pointer-events="none"
          text-anchor="middle"
          width="200"
          x="100"
          y="50"
        >
          <tspan
            dy="5.5"
            x="100"
          >
            Data Object
          </tspan>
        </text>
      </g>
    </svg>
  </div>
</body>
`;
